src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Glue;
+using Microsoft.PowerFx.Core.Lexer;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Core.Public;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Syntax;
+using Microsoft.PowerFx.Core.Texl.Intellisense;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Expose binding logic for Power Fx. 
+    /// Derive from this to provide evaluation abilities. 
+    /// </summary>
+    public class Engine : IPowerFxEngine
+    {
+        /// <summary>
+        /// Configuration symbols for this Power Fx engine.
+        /// </summary>
+        public PowerFxConfig Config { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Engine"/> class.
+        /// </summary>
+        /// <param name="powerFxConfig"></param>
+        public Engine(PowerFxConfig powerFxConfig)
+        {
+            powerFxConfig.Lock();
+            Config = powerFxConfig;
+        }
+
+        /// <summary>
+        /// List all functions (both builtin and custom) registered with this evaluator. 
+        /// </summary>
+        public IEnumerable<string> GetAllFunctionNames()
+        {
+            return Config.GetAllFunctionNames();
+        }
+
+        /// <summary>
+        /// Create a resolver for use in binding. This is called from <see cref="Check(string, FormulaType)"/>.
+        /// Base classes can override this is there are additional symbols not in the config.
+        /// </summary>
+        /// <param name="parameterType"></param>
+        /// <returns></returns>
+        private protected virtual SimpleResolver CreateResolver()
+        {
+            return new SimpleResolver(Config);
+        }
+
+        /// <summary>
+        /// Type check a formula without executing it. 
+        /// </summary>
+        /// <param name="expressionText"></param>
+        /// <param name="parameterType"></param>
+        /// <returns></returns>
+        public CheckResult Check(string expressionText, RecordType parameterType = null)
+        {
+            if (parameterType == null)
+            {
+                parameterType = new RecordType();
+            }
+
+            var formula = new Formula(expressionText);
+
+            formula.EnsureParsed(TexlParser.Flags.None);
+
+            // Ok to continue with binding even if there are parse errors. 
```

Comment:
It might be useful to add a flag that would stop here and just report errors. In some context that might be useful for performance purposes - for example, when binding is not important if there are already syntax errors. However, this does not seem super useful right now.